synapse/storage/databases/main/devices.py

```diff
@@ -1130,6 +1130,12 @@ async def delete_device(self, user_id: str, device_id: str) -> None:
             desc="delete_device",
         )
 
+        await self.db_pool.simple_delete_one(
```

Comment:
```suggestion
        await self.db_pool.simple_delete_many(
```

Since the table may have many rows (or no rows)